[MIG] partner_financial_risk : Migration to 10.0

chienandalu opened the pull request:
- Superseeding https://github.com/OCA/partner-contact/pull/451 in order to finish PR.
- Pick of https://github.com/OCA/partner-contact/pull/468/commits/38ca7054c83271ee3512c6ce4c6a5bcc3f6d39f7
- Some minor changes in README.rst added

carlosdauden committed:
[ADD] partner_financial_risk (#292)

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[9.0][FIX] partner_financial_risk: Refund were not contemplated

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[9.0][IMP] partner_financial_risk: Improve performance (#385)

* [9.0][IMP] partner_financial_risk: Improve performance

* [9.0][IMP] partner_financial_risk: Improve tests and partner hierarchy

* [9.0][IMP] partner_stock_risk: Test

oca-transbot committed:
OCA Transbot updated translations from Transifex

Ehtaga committed:
[MIG] Partner Financial Risk : Migration to 10.0

carlosdauden committed:
[9.0][FIX] partner_financial_risk: Error when risk_total includes draft_invoices

chienandalu committed:
[IMP] Minor changes

pedrobaeza commented:
It seems that Travis gets stalled...

pedrobaeza commented:
Try to change test to TransactionCase o HttpCase to see if it makes it to finish Travis

carlosdauden commented on partner_financial_risk/__manifest__.py:
```diff
@@ -5,17 +5,19 @@
-    'depends': ['account'],
+    'depends': [
+        'account',
+        'sales_team',
```
I think that this dependendy is not needed.
You can remove 'base.group_sale_manager' and keep only 'account.group_account_manager'.
That group was because depends of base.

carlosdauden commented on partner_financial_risk/models/res_partner.py:
```diff
@@ -60,7 +60,7 @@ class ResPartner(models.Model):
     def _compute_risk_allow_edit(self):
         is_editable = self.env.user.has_group(
-            'base.group_sale_manager') or self.env.user.has_group(
+            'sales_team.group_sale_manager') or self.env.user.has_group(
```
Remove and add 'sales_team.group_sale_manager' group in partner_sale_risk

carlosdauden commented on partner_financial_risk/tests/test_partner_financial_risk.py:
```diff
@@ -2,15 +2,15 @@
     def setUpClass(cls):
         super(TestPartnerFinancialRisk, cls).setUpClass()
-        cls.env.user.groups_id |= cls.env.ref('base.group_sale_manager')
+        cls.env.user.groups_id |= cls.env.ref('sales_team.group_sale_manager')
```
Test with 'account.group_account_manager'

carlosdauden commented on partner_financial_risk/views/res_partner_view.xml:
```diff
@@ -2,13 +2,15 @@
         <field name="inherit_id" ref="base.view_partner_form" />
         <field name="priority" eval="20"/>
-        <field name="groups_id" eval="[(4, ref('base.group_sale_salesman')), (4, ref('account.group_account_manager'))]"/>
+        <field name="groups_id" eval="[(4, ref('sales_team.group_sale_manager')), (4, ref('account.group_account_manager'))]"/>
```
Remove (4, ref('sales_team.group_sale_manager')),
Append in partner_sale_risk

carlosdauden commented on partner_financial_risk/wizard/partner_risk_exceeded_view.xml:
```diff
@@ -17,7 +17,7 @@
                             name="button_continue"
                             type="object"
-                            groups="base.group_sale_manager,account.group_account_manager"
+                            groups="sales_team.group_sale_manager,account.group_account_manager"
```
Remove sales_team.group_sale_manager.
Append in partner_sale_risk

pedrobaeza commented:
Please cherry-pick https://github.com/OCA/partner-contact/commit/63abcfb51909d76dfd2add7342258fb23c9cd4a3

chienandalu commented:
Yes, @pedrobaeza, it's already in the history: https://github.com/OCA/partner-contact/pull/470/commits/44fecc2ea2ce206871295d8016ee2a1a45f8064d

chienandalu commented:
Could it be that Travis is having issues with thist test comments?:

```
2017-09-08 10:34:31,131 4870 INFO openerp_test odoo.addons.partner_alias.tests.test_partner_alias: ` Test raise ValidationError if alias name same as partner
2017-09-08 10:34:31,154 4870 INFO openerp_test odoo.addons.partner_alias.tests.test_partner_alias: test_unique_name (odoo.addons.partner_alias.tests.test_partner_alias.TestPartnerAlias)
2017-09-08 10:34:31,154 4870 INFO openerp_test odoo.addons.partner_alias.tests.test_partner_alias: ` Test raises IntegrityError is name not unique
2017-09-08 10:34:31,159 4870 INFO openerp_test odoo.sql_db: bad query: UPDATE "res_partner_alias" SET "firstname"='Greg',"write_uid"=1,"write_date"=(now() at time zone 'UTC') WHERE id IN (1)
2017-09-08 10:34:31,160 4870 INFO openerp_test odoo.addons.partner_alias.tests.test_partner_alias: Ran 2 tests in 0.030s
```

pedrobaeza commented:
Make a try of excluding that module in the .travis.yml file to see if it works

chienandalu commented:
Well, I've commited without tests and travis succeeds so it's clear now where the problem is

chienandalu committed:
[9.0][IMP] partner_financial_risk: Add track visibility to credit limit

rafaelbn commented:
Hi @chienandalu please take care to bring all fixes from https://github.com/OCA/partner-contact/pull/476 here too 😄 thanks!

carlosdauden committed:
[FIX] partner_financial_risk: Invoice refunds add risk

- Refunds adds partner risk instead of reduce.
- If date_maturity changes to previous today date "never" compute partner risk.

carlosdauden committed:
[IMP] partner_financial_risk: New fields in view. Improve code.

rousseldenis commented:
@chienandalu Can you check travis ? It seems your tests fail.

carlosdauden committed:
[10.0][IMP] partner_financial_risk: Improve multicompany cron (#530)

* [9.0][IMP] partner_financial_risk: Improve multicompany cron

* [9.0][IMP] partner_financial_risk: Exec with sudo

(cherry picked from commit 6590bbf)

pedrobaeza commented:
@chienandalu please check Travis and review latest commits and PR from 9.0 that have new features.

rafaelbn commented:
Tried to test in runbut, but Financial Risk tab is wrong, check boxes and fields doesn't work

Please check screenshot

![2018-04-27_0-23-42](https://user-images.githubusercontent.com/8736623/39334989-6b7f3460-49b1-11e8-8698-2a0838a65aa2.png)

rafaelbn reviewed (changes requested):
Please @chienandalu fix, thanks!

chienandalu committed:
[FIX] partner_financial_risk: risk table layout

chienandalu commented:
Tests keep crashing on travis:
```
odoo.addons.partner_financial_risk.tests.test_partner_financial_risk: ` InternalError: no such savepoint
```

chienandalu commented:
@rafaelbn Check the partner risk table now

rafaelbn commented:
Check @chienandalu 

![2018-05-09_14-18-19](https://user-images.githubusercontent.com/8736623/39814297-0f8ec4b2-5394-11e8-86b9-c6efed678a10.png)

rafaelbn commented:
@chienandalu did you attend @carlosdauden comments? https://github.com/OCA/partner-contact/pull/470#pullrequestreview-61431471

chienandalu commented:
@rafaelbn Yes, those comment were attended. Take a look at it now.

chienandalu committed:
[FIX] tests get stalled on travis

chienandalu commented:
.travis is still failing. Seems to be unable to make the rollback on a SavepointCase. @Yajo @moylop260 Any idea of why this could be happening? It doesn't happen in other branches/repos...

moylop260 commented:
@chienandalu I have reproduced this issue from a local docker container using travis2docker...

If I comment the following line:
 - [10.0 - partner_changeset/models/res_partner.py#L28-L39::write](https://github.com/OCA/partner-contact/blob/7cabfe2067ca92e314268d5e892312556e954525/partner_changeset/models/res_partner.py#L28-L39)

The first error is not showed
```bash
self.assertTrue(self.partner.risk_exception)
odoo.addons.partner_financial_risk.tests.test_partner_financial_risk: ` AssertionError: False is not true
```

If we not install `partner_identification`  modules then the second error `savepoint rollback` is not showed. (Notice that `partner_coc` depends of `partner_identification` then exclude it too).

If I exclude the `EXCLUDE=partner_changeset,partner_identification,partner_coc` modules then run all tests green

I have created a dummy commit in your branch in order to see my hypothesis and you can verify the matter and fix it.

You can reproduce them locally running:
 1. Install depends modules: 
   - `~/odoo-10.0/odoo-bin -d db_test --db-filter=db_test --stop-after-init -i partner_financial_risk`
 2. Error 1:
   - `~/odoo-10.0/odoo-bin -d db_test --db-filter=db_test --stop-after-init -i partner_financial_risk,partner_changeset --test-enable`
 3. Error 2: 
   - `~/odoo-10.0/odoo-bin -d db_test --db-filter=db_test --stop-after-init -i partner_financial_risk,partner_identification --test-enable`

chienandalu commented:
Wow, thanks @moylop260 I thought I was never to see this green!

@rafaelbn @carlosdauden Can you review again?

moylop260 commented:
FYI the commit is a test...
IMHO It Shoudn’t be the solution

-- 
Moisés López Calderón
Mobile: (+521) 477-752-22-30
Twitter: @moylop260
hangout: moylop260@vauxoo.com
http://www.vauxoo.com - Odoo Gold Partner
Twitter: @vauxoo

chienandalu commented:
@moylop260  Sure, I know! Thanks!

pedrobaeza commented:
Is this finished then?

chienandalu commented:
@pedrobaeza Well, as @moylop260 found out, there are two modules in this repoitory that clash along with this one (partner_changeset and partner_identification). Maybe there are two options here:

- The quick one could be to move this PR into the new repo in https://github.com/OCA/credit-control and write down in known issues these problematic interactions
- Or we could try and debug where these three are clashing

chienandalu committed:
[FIX] partner-contact CI interactions

chienandalu commented:
Green! :four_leaf_clover:

rafaelbn reviewed (approved):
Tested functionally :+1:
